### PR TITLE
feat: implement jpeg xl codec

### LINKED ISSUE #79: Implement JPEG XL Codec
Find a way to encode/decode JPEG XL images.

I found two ways to do it:

### Pros
* Reference implementation
* Feature packed
* Multi-threaded encoding/decoding
### Cons
* C API, we'll have to generate bindings to it using `cbindgen`, that means lots of raw pointers and fun
* Build system, bundling non-Rust libraries is a PITA

### Pros
* Rust API/implementation
### Cons
* Limited API
* Single-threaded

### PR DESCRIPTION
**Pull request by AndreasLrx**
Fix default max image size (10_000_000 -> 20_000_000)
And openapi oauth2 urls

### CODE REVIEW

**Review by BaptistePgn — DISMISSED** [review]

File: back/Cargo.toml [hunk 1]
```diff
@@ -57,6 +57,7 @@ image-backend = { version = "~0.25.0", package = "image", default-features = fal
     "gif",
 ] }
 md-5 = "0.10.6"
+jpegxl-rs = { version = "~0.10.2", features=["image"] }
```
Comment (MisterPeModder) [inline back/Cargo.toml]: nit: formatting; should be `features = [ "image" ]`

File: back/src/api/images.rs
Comment (MisterPeModder) [inline back/src/api/images.rs:1]: The algorithm at line [423](https://github.com/tistech0/Pictou/pull/101/files#diff-3e0c03cce2926a64d6ee317f1f2b45a99c3c8c2f386d66b8787815ce8d0674e2R423) should be "jxl" or something.

File: README.md [hunk 2]
```diff
@@ -15,6 +15,8 @@ cd PROJECT_DIR
 pre-commit install
 ```
 
+Install [libjxl 10.0.2](https://github.com/libjxl/libjxl/releases/tag/v0.10.2) on your system for the back.
+
```
Comment (MisterPeModder) [inline README.md:19]: Pkg-config is also needed for the build on Linux. For windows and MacOS I don't know.

**Review by MisterPeModder — CHANGES_REQUESTED** [review]
Works pretty well for real life images !

For some reasons, the following image fails to encode properly and takes 4x the storage space when "compressed" (from 143kB to 451kB).

sbt_banner

Also here's a patch to get the build working under Nix:
(apply with `patch shell.nix < thepatch.patch`)
```patch
--- a/shell.nix
+++ b/shell.nix
@@ -1,14 +1,32 @@
 { pkgs ? import <nixpkgs> {} }:

-pkgs.mkShell {
-  name = "pictou";
-  buildInputs = with pkgs; [
-    pre-commit
-    diesel-cli
-    postgresql
-  ];
+let
+  libjxl-0-10-2 = pkgs.libjxl.overrideAttrs (old: {
+    version = "0.10.2";
+    src = pkgs.fetchFromGitHub {
+      owner = "libjxl";
+      repo = "libjxl";
+      rev = "v0.10.2";
+      hash = "sha256-Ip/5fbzt6OfIrHJajnxEe14ppvX1hJ1FSJUBEE/h5YQ=";
+      # There are various submodules in `third_party/`.
+      fetchSubmodules = true;
+    };
+  });
+in
+  pkgs.mkShell {
+    name = "pictou";
+    buildInputs = with pkgs; [
+      pre-commit
+      diesel-cli
+      postgresql
+      libjxl-0-10-2
+    ];

-  shellHook = ''
-    pre-commit install
-  '';
-}
+    nativeBuildInputs = with pkgs; [
+      pkg-config
+    ];
+
+    shellHook = ''
+      pre-commit install
+    '';
+  }
```

**Review by MisterPeModder — APPROVED** [review]

**Pull request merged**